client/flutter/lib/api/display_conditions.dart

```diff
@@ -1,9 +1,77 @@
 import 'package:expressions/expressions.dart';
 import 'package:meta/meta.dart';
 import 'package:flutter/material.dart';
+import 'package:who_app/api/user_preferences.dart';
 
 import 'package:who_app/pages/symptom_checker/symptom_checker_model.dart';
 
+class LogicContext {
+  /// alpha-2 code
+  final String isoCountryCode;
+
+  /// Consistent value in [0-100) for experiment arms
+  final int cohort;
+
+  /// ISO 8601 date and time, ex 2017-09-05T14:49Z
+  final String isoDateTimeUTC;
+
+  /// Local days completed since Jan 1, 2020
+  final int localDays;
+
+  LogicContext(
+      {@required this.isoCountryCode,
+      @required this.cohort,
+      @required this.isoDateTimeUTC,
+      @required this.localDays});
+
+  static final _refDate = DateTime(2020);
+
+  static Future<LogicContext> generate() async {
+    final now = DateTime.now().toUtc();
+    final diff = now.toLocal().difference(_refDate).inDays;
+    return LogicContext(
+      cohort: await UserPreferences().getCohort(),
+      isoCountryCode: await UserPreferences().getCountryIsoCode(),
```

Comment:
Will this be a problem if the expression evaluator attempts to evaluate something that is null?

Reply:
Null works in the expressions package. 👍